eth/api.go

```diff
@@ -346,6 +347,47 @@ func (api *PrivateDebugAPI) GetBadBlocks(ctx context.Context) ([]*BadBlockArgs,
 	return results, nil
 }
 
+type addressMap map[common.Hash]common.Address
+
+type AccountRangeResult struct {
+	AddressMap addressMap `json:"addressMap"`
+}
+
+func accountRange(st state.Trie, start *common.Hash, maxResult int) (AccountRangeResult, error) {
+	it := trie.NewIterator(st.NodeIterator(start[:]))
+	result := AccountRangeResult{AddressMap: addressMap{}}
+	for i := 0; i < maxResult && it.Next(); i++ {
+		key := st.GetKey(it.Key)
+		// If key is nil, that means it wasn't found in the preimage database.
+		// This is not a problem, because we still return the hash of the key together with
+		// address zero and the client can very easily determine that the hash of the address zero
+		// is not matching, which means it wasn't found.
+		result.AddressMap[common.BytesToHash(it.Key)] = common.BytesToAddress(key)
+	}
+	return result, nil
+}
+
+//block hash or number, tx index, start address hash, max results
+func (api *PrivateDebugAPI) AccountRangeAt(ctx context.Context, blockNr rpc.BlockNumber, txIndex int, startAddr *common.Hash, maxResults int) (AccountRangeResult, error) {
```

Comment:
I don't really see what the point of blockNr and txIndex is. Why not just pass in the transaction hash like the other API methods and figure out the rest internally?

Reply:
The "iterator approach" of limiting the results is also probably a bad idea, because then you need to regenerate the execution environment for every call.

Reply:
There's also a high chance that the parent state is not available any more, in which case you'd need to first regenerate it similarly to tracing.

Reply:
I believe this is the way `aleth` implemented it. @winsvega any reasons against changing it as proposed by @karalabe  ?

Reply:
For our purposes, we can limit this to be only able to enumerate the accounts of the state at the latest block.

Reply:
> I don't really see what the point of blockNr and txIndex is. Why not just pass in the transaction hash like the other API methods and figure out the rest internally?

Then I have no way of seeing the updated list of accounts if there have been no transactions recently.